tests/v3/test_store/test_stateful_store.py

```diff
@@ -0,0 +1,190 @@
+# Stateful tests for arbitrary Zarr stores.
+
+import asyncio
+
+import hypothesis.strategies as st
+from hypothesis import note
+from hypothesis.stateful import (
+    RuleBasedStateMachine,
+    invariant,
+    precondition,
+    rule,
+)
+
+from zarr.abc.store import Store as StoreABC
+from zarr.buffer import Buffer, default_buffer_prototype
+from zarr.store import LocalStore, MemoryStore, RemoteStore
+
+# from strategies_store import StoreStatefulStrategies, key_ranges
+from zarr.testing.strategies import key_ranges, paths
+
+# zarr spec: https://zarr-specs.readthedocs.io/en/latest/v3/core/v3.0.html
+
+
+class SyncMemStoreWrapper(StoreABC):
```

Comment:
```suggestion
class SyncMemStoreWrapper():
```

Sorry I led you down the wrong path with the ABC suggestion, it's an async ABC which doesn't help us. Just make sure the method names match for convenience.